src/darwin/Framework/CHIP/MTRDeviceControllerDataStore.mm

```diff
@@ -151,6 +151,17 @@ - (nullable instancetype)initWithController:(MTRDeviceController *)controller
     return self;
 }
 
+- (void)startUpWithClusterDataHandler:(MTRDeviceControllerDataStoreClusterDataHandler)clusterDataHandler
+{
+    dispatch_sync(_storageDelegateQueue, ^{
+        if ([self->_storageDelegate respondsToSelector:@selector(valuesForController:securityLevel:sharingType:)]) {
+            NSDictionary<NSString *, id> * dataStoreSecureLocalValues = [self->_storageDelegate valuesForController:self->_controller securityLevel:MTRStorageSecurityLevelSecure sharingType:MTRStorageSharingTypeNotShared];
+
+            clusterDataHandler([self getClusterDataFromSecureLocalValues:dataStoreSecureLocalValues]);
```

Comment:
So this calls `clusterDataHandler` while running on the delegate queue.  That seems a bit risky to me; what migt be better is if we do the dispatch_sync around just the `valuesForController` call, get the results into a `__block` `NSDictionary *`, and then do the `getClusterDataFromSecureLocalValues` and calling of the handler back on the original queue we got called on.

Reply:
Done.